### PR TITLE
Add testing of bam_processing and  models

### PR DESCRIPTION
**Pull request by sndrtj**

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 pytest>=4.1.0
 flake8>=3.6.0
-pytest-cov>=2.6.1
\ No newline at end of file
+pytest-cov>=2.6.1
+python-magic>=0.4.15
```
Comment (rhpvorderman) [inline requirements-dev.txt:4]: I love [magic](https://www.youtube.com/watch?v=MzlK0OGpIRs)!

File: rna_cd/bam_process.py [hunk 2]
```diff
@@ -124,6 +124,8 @@ def make_array_set(bam_files: List[Path], labels: List[Any],
     :return: tuple of X and Y numpy arrays. X has shape (n_files, n_features)
     Y has shape (n_files,).
     """
+    if cores < 1:
+        raise ValueError("Number of cores must be at least 1.")
```
Comment (rhpvorderman) [inline rna_cd/bam_process.py:128]: :+1:

File: rna_cd/models.py [hunk 3]
```diff
@@ -72,6 +72,10 @@ def train_svm_model(positive_bams: List[Path], negative_bams: List[Path],
         raise ValueError("The list of positive BAM files may not be empty.")
     if len(negative_bams) < 1:
         raise ValueError("The list of negative BAM files may not be empty.")
+
+    if set(positive_bams) == set(negative_bams):
+        raise ValueError("Positive and negative BAM files may not be "
+                         "identical.")
```
Comment (rhpvorderman) [inline rna_cd/models.py]: This is not entirely the metric that you want I guess.
There may not be any negative bams that are also listed as positive bams and vice versa.
```python
if  not set(positive_bams).isdisjoint(set(negative_bams)):
    raise ValueError("bam files were given that occurred in both the positive and negative set")
```
[`.isdisjoint`](https://docs.python.org/3.5/library/stdtypes.html#frozenset.isdisjoint) checks whether two sets have no elements in common. 

If you are nice you will also give them the list of these files using set's intersection method.

**Review by rhpvorderman — APPROVED** [review]
Looks good just one comment on the content. But I think it is important that it is addressed.
I approve for now so you can get on with work after the fix.

File: rna_cd/models.py (hunk 3 above)
Comment (sndrtj) [reply rna_cd/models.py]: Very good point!

**Pull request merged**